File: 11-container-with-most-water/cpp/main.cpp
class Solution {
 public:
  int maxArea(vector<int>& height) {
    int l = 0, r = height.size() - 1, maxHeight = 0;

    while (l < r) {
      maxHeight = max(maxHeight, (r - l) * min(height[l], height[r]));

      if (height[l] < height[r]) {
        ++l;
      } else {
        --r;
      }
    }

    return maxHeight;
  }
};

File: 11-container-with-most-water/python/main.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i, j = 0, len(height) - 1
        max_water = -1
        while i < j:
            max_water = max(max_water, (j - i) * min(height[i], height[j]))

            if height[i] < height[j]:
                i += 1
            else:
                j -= 1

        return max_water
